pkg/operation/botanist/resource_manager.go

```diff
@@ -101,3 +159,110 @@ func (b *Botanist) DeployGardenerResourceManager(ctx context.Context) error {
 func (b *Botanist) ScaleGardenerResourceManagerToOne(ctx context.Context) error {
 	return kubernetes.ScaleDeployment(ctx, b.K8sSeedClient.Client(), kutil.Key(b.Shoot.SeedNamespace, v1beta1constants.DeploymentNameGardenerResourceManager), 1)
 }
+
+func (b *Botanist) mustBootstrapGardenerResourceManager(ctx context.Context) (bool, error) {
+	shootAccessSecret := gutil.NewShootAccessSecret(resourcemanager.SecretNameShootAccess, b.Shoot.SeedNamespace)
+	if err := b.K8sSeedClient.Client().Get(ctx, client.ObjectKeyFromObject(shootAccessSecret.Secret), shootAccessSecret.Secret); err != nil {
+		if !apierrors.IsNotFound(err) {
+			return false, err
+		}
+		return true, nil // Shoot access secret does not yet exist.
+	}
+
+	renewTime, err2 := time.Parse(time.RFC3339, shootAccessSecret.Secret.Annotations[resourcesv1alpha1.ServiceAccountTokenRenewTimestamp])
+	if err2 != nil {
+		return false, fmt.Errorf("could not parse renew timestamp: %w", err2)
+	}
+	if time.Now().UTC().After(renewTime.UTC()) {
+		return true, nil // Shoot token was not renewed.
+	}
+
+	managedResource := &resourcesv1alpha1.ManagedResource{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      resourcemanager.ManagedResourceName,
+			Namespace: b.Shoot.SeedNamespace,
+		},
+	}
+
+	if err := b.K8sSeedClient.Client().Get(ctx, client.ObjectKeyFromObject(managedResource), managedResource); err != nil {
+		if !apierrors.IsNotFound(err) {
+			return false, err
+		}
+		return true, nil // ManagedResource (containing the RBAC resources) does not yet exist.
+	}
+
+	if conditionApplied := v1beta1helper.GetCondition(managedResource.Status.Conditions, resourcesv1alpha1.ResourcesApplied); conditionApplied != nil &&
+		conditionApplied.Status == gardencorev1beta1.ConditionFalse &&
+		strings.Contains(conditionApplied.Message, `forbidden: User "system:serviceaccount:kube-system:gardener-resource-manager" cannot`) {
```

Comment:
Not a requirement for this PR but we should definitely make use of an error code for this case. String matching is rather brittle.

Reply:
Yeah, for now it's precise and good enough, I think (all other solutions would increase the complexity with potentially little value at this point time).

Reply:
Hm, in general I cannot agree here that it brings little value. Yes, it works for now but it's super susceptible to any changes that lead to the message not being as it's expected. Such messages are typically not seen as part of an API contract, e.g. in some months a developer decides to not expose the API response in the `conditionApplied.Message`. 

> all other solutions would increase the complexity

It should be straightforward: If GRM notices a `403` it'll assign a proper error code which can be re-used here.

I still don't think it's a absolute requirement for this PR but we should really avoid any string matching where possible to safe us unexpected trouble in the future.